--rm-kms panics when multiple KMS from the same ARN need to be removed

I have this file that has 3 KMS entries, 2 of which are from the same ARN. When calling `--rm-kms` with the ARN that has 2 entries, sops panics:

```
$ sops --rm-kms arn:aws:kms:eu-central-1:36******:key/cb1fab90-8d17-42a1-a9d8-334968904f94 -r -i somefile.txt
panic: runtime error: slice bounds out of range [recovered]
	panic: runtime error: slice bounds out of range

goroutine 1 [running]:
panic(0x859240, 0xc4200121b0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
go.mozilla.org/sops/vendor/gopkg.in/urfave/cli%2ev1.HandleAction.func1(0xc4200ff7f8)
	/home/ulfr/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:472 +0x29e
panic(0x859240, 0xc4200121b0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
go.mozilla.org/sops.(*Metadata).RemoveMasterKeys(0xc4200ff1d8, 0xc42022b600, 0x1, 0x1)
	/home/ulfr/src/go.mozilla.org/sops/sops.go:275 +0x313
go.mozilla.org/sops.(*Metadata).RemoveKMSMasterKeys(0xc4200ff1d8, 0xc4203e34f0, 0x4e)
	/home/ulfr/src/go.mozilla.org/sops/sops.go:340 +0x16e
main.rotate(0xc420086c80, 0x7ffc165f5cab, 0x18, 0xc420110000, 0x12f8, 0x14f8, 0xad12a0, 0xc4200320e0, 0x0, 0x18)
	/home/ulfr/src/go.mozilla.org/sops/cmd/sops/main.go:414 +0x825
main.main.func1(0xc420086c80, 0x0, 0x0)
	/home/ulfr/src/go.mozilla.org/sops/cmd/sops/main.go:196 +0x413
reflect.Value.call(0x840440, 0x90b980, 0x13, 0x8cb747, 0x4, 0xc4200ff798, 0x1, 0x1, 0x4c6e28, 0x8c06c0, ...)
	/usr/local/go/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0x840440, 0x90b980, 0x13, 0xc4200ff798, 0x1, 0x1, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:302 +0xa4
go.mozilla.org/sops/vendor/gopkg.in/urfave/cli%2ev1.HandleAction(0x840440, 0x90b980, 0xc420086c80, 0x0, 0x0)
	/home/ulfr/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:481 +0x1e0
go.mozilla.org/sops/vendor/gopkg.in/urfave/cli%2ev1.(*App).Run(0xc420104000, 0xc42000c120, 0x6, 0x6, 0x0, 0x0)
	/home/ulfr/src/go.mozilla.org/sops/vendor/gopkg.in/urfave/cli.v1/app.go:240 +0x596
main.main()
	/home/ulfr/src/go.mozilla.org/sops/cmd/sops/main.go:200 +0xd5a
```

--- a/sops.go
+++ b/sops.go
@@ -261,14 +261,20 @@ func (m *Metadata) MasterKeyCount() int {
 // RemoveMasterKeys removes all of the provided keys from the metadata's KeySources, if they exist there.
 func (m *Metadata) RemoveMasterKeys(keys []MasterKey) {
 	for j, ks := range m.KeySources {
-		for i, k := range ks.Keys {
-			for _, k2 := range keys {
-				if k.ToString() == k2.ToString() {
-					ks.Keys = append(ks.Keys[:i], ks.Keys[i+1:]...)
+		var newKeys []MasterKey
+		for _, k := range ks.Keys {
+			matchFound := false
+			for _, keyToRemove := range keys {
+				if k.ToString() == keyToRemove.ToString() {
+					matchFound = true
+					break
 				}
 			}
+			if !matchFound {
+				newKeys = append(newKeys, k)
+			}
 		}
-		m.KeySources[j] = ks
+		m.KeySources[j].Keys = newKeys
 	}
 }
 

--- a/sops_test.go
+++ b/sops_test.go
@@ -4,6 +4,7 @@ import (
 	"bytes"
 	"github.com/stretchr/testify/assert"
 	"go.mozilla.org/sops/aes"
+	"go.mozilla.org/sops/kms"
 	"reflect"
 	"testing"
 )
@@ -200,4 +201,37 @@ func TestTruncateTree(t *testing.T) {
 	result, err := tree.Truncate(`["bar"]["foobar"][2]`)
 	assert.Equal(t, nil, err)
 	assert.Equal(t, expected, result)
+}
+
+func TestRemoveMasterKeys(t *testing.T) {
+	m := Metadata{
+		KeySources: []KeySource{
+			KeySource{
+				Name: "kms",
+				Keys: []MasterKey{
+					&kms.MasterKey{
+						Arn: "foo",
+					}, &kms.MasterKey{
+						Arn: "bar",
+					},
+					&kms.MasterKey{
+						Arn: "foobar",
+					},
+				},
+			},
+		},
+	}
+	m.RemoveMasterKeys([]MasterKey{
+		&kms.MasterKey{
+			Arn: "bar",
+		},
+		&kms.MasterKey{
+			Arn: "foobar",
+		},
+	})
+	assert.Equal(t, []MasterKey{
+		&kms.MasterKey{
+			Arn: "foo",
+		},
+	}, m.KeySources[0].Keys)
 }

